packages/components/src/components/Select/useSelectData.tsx

```diff
@@ -27,12 +42,52 @@ export const getRecursiveOptionValues = (children: ReactNode): { value: string;
     return values;
 };
 
-export const useSelectData = (children: ReactNode) => {
+/**
+ * Custom hook for managing select data and filtering.
+ * This hook processes the children to extract select items, handles filtering,
+ * and manages the state for custom values if allowed.
+ *
+ * @param {ReactNode} children - The React children to process, typically SelectItem components.
+ * @param {boolean} [allowCustomValue=false] - Whether to allow custom values that are not in the predefined list.
+ * @returns {Object} An object containing various select data and utility functions.
+ * @property {ReactNode[]} inputSlots - Slots for input customization (left and right).
+ * @property {ReactNode[]} menuSlots - Slots for menu customization.
+ * @property {ReactNode} clearButton - The clear button component if provided.
+ * @property {string} filterText - The current filter text.
+ * @property {SelectItemType[]} items - The filtered list of items based on the current filter text.
+ * @property {boolean} valueExists - Whether the current filter text matches an existing item.
+ * @property {SelectItemType | undefined} existingValueItem - The matching item if valueExists is true.
+ * @property {boolean} shouldAddCustomItem - Whether a custom item should be added based on the current state.
+ * @property {boolean} valueInvalid - Whether the current value is invalid (when custom values are not allowed).
+ * @property {function} setFilterText - Function to update the filter text.
+ * @property {function} getItemByValue - Function to get an item by its value.
+ */
+export const useSelectData = (children: ReactNode, allowCustomValue = false) => {
     const [filterText, setFilterText] = useState('');
-    const { inputSlots, menuSlots, itemValues, clearButton } = useMemo(() => {
+
+    const itemValues = useMemo(() => getRecursiveOptionValues(children), [children]);
```

Comment:
this will most likely cause issues, as `children` also include the Input Slots. If slots are present, we only pass menuSlots